contents/docker/Docker.md

```diff
@@ -0,0 +1,161 @@
+<frontmatter>
+  title: DevOps
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Docker
+
+Authors: [Rahul Rajesh](https://github.com/rrtheonlyone)
+
+### 1. What is Docker?
+
+Docker is a platform that is used to develop, deploy and run applications inside containers. This article will discuss some of the key ideas behind Docker, the reasons to use Docker and some resources to guide you in this process.
+
+#### 1.1 Virtual Machine (VM)
+
+Before the rise of Docker, people used VMs. A VM is an emulation of a real computer that executes program like a real computer. A VM is used to isolate an application and its dependancies into a self-contained unit that can run anywhere.
+
+![alt text](./Virtual_Machine.png "VM Diagram")
+
+With reference to the above diagram, VMs run on top of a physical machine using a “hypervisor”. A VM is able to provide virtual hardware, a kernel (i.e. OS) and user space. A Windows user can use a VM to run an application that is only supported on OSX for example. 
+
```

Comment:
This is a very general introduction to container, people not familiar with OS might find it hard to understand(For example, How does it packages up user space?) Probably provides some links for further study of how container works(Liked what I have presented). Probably mention that docker utilizes OS-level support(namespaces, cgroups) and provides link to it